docs/release-strategy.md

```diff
@@ -0,0 +1,83 @@
+# InstructLab Eval Release Strategy
+
+This document discusses the release strategy and processes for the
+`instructlab-eval` Python package built from the
+<https://github.com/instructlab/eval> git repository.
+
+## Versioning Scheme
+
+Releases use a `X.Y.Z` numbering scheme.
+
+X-stream release are for major releases. At this stage in the project a major release has not been cut and we expect each release to be a new Y-stream.
+
+Z-stream releases are meant for critical bug and documentation fixes. Z-stream releases are cut as maintainers see fit.
+
+## Schedule
+
+The project currently operates on an ad-hoc release schedule based on the discretion of the maintainers team.
+
+The cadence for major releases starting from 1.0 onward will be determined as the project matures.
+
+A schedule will be updated in a markdown file on the <https://github.com/instructlab/eval> GitHub repository.
+
+## Release Tracking
+
+Currently there is no former process of release tracking. GitHub Issues are used for tracking individual work items.
+
+In the future, the project may use Milestones or Project Boards for more formal release planning. At that time this document will be updated.
+
+## Git Branches and Tags
+
+Every `X.Y` release stream gets a new branch.
+
+Each release, `X.Y.Z`, exists as a tag named `vX.Y.Z`.
+
+## Release Branch Maintenance
+
+Maintenance efforts are only on the most recent Y-stream.
+Critical bug fixes are backported to the most recent release branch.
+
+## Release Mechanics
+
+Release mechanics are done by a Release Manager identified for that release.
+The Release Manager is a member of the Eval Maintainers team that has agreed to take on these responsibilities.
+The Release Manager can change on a per-release basis.
+
+The following are the steps for how Y-stream and Z-stream releases gets cut.
+
+### Y-Stream
+
+1. Determine a commit on the main branch that will serve as the basis for the next release - most of the time this should be the latest commit.
+1. Create a new release branch in the format `release-vX.Y` off of the determined commit (will match `main` if the latest commit is chosen).
+1. Create a new release on GitHub targeting the release branch and using the latest Y-Stream tag as the previous release (e.g. `0.15.1` precedes `0.16.0`).
+1. Announce release via the following:
+    - The `#eval` channel on Slack
+    - The `dev` mailing list
+
+### Z-Stream
+
+1. Backport all relevant commits from `main` to the `release-vX.Y` branch.
```

Comment:
Noting this can also happen in the other direction.  Fix is developed for the release branch and needs to be ported to main.

Maybe state as ensure corresponding fixes are made to the release stream as well as main and any release streams between those two versions.

Reply:
Hmmm, in CLI, all fixes go into `main` first and then are backported. When would the reverse occur?

Reply:
It depends on the driver, if we find the issue on main, we would typically backport as needed.  If the issue is reported by a user on a particular version, you might recreate and fix it there first to get that user enabled.  Especially if the work was very different between the two versions.

Reply:
I feel like if update that here, we should do the same in the CLI. AFAIK we are only supporting the most recent Y-Stream at this juncture. So we can do that update in both, but I'm not sure if it'll come into play until we start supporting multiple versions.

Reply:
For our v0.1.2 release, I ended up merging to the release branch first:

https://github.com/instructlab/eval/pull/110

and then backporting to main:

https://github.com/instructlab/eval/pull/112

In this case I did it because we wanted to get the fix out asap and didn't want to wait for tests on main, then backport and wait for test on the release branch.

Reply:
Very good point - I'm going to update the doc with some wording for that - once that's in I think we could go for one last round of reviews and merge this